src/features/projectsV2/components/microComponents/ProjectBadge.tsx

```diff
@@ -0,0 +1,105 @@
+
+interface Props {
+  isApproved: boolean;
+  isTopProject: boolean;
+  allowDonations: boolean;
+}
+
+interface TitleAndIconReturnType {
+  icon: ReactElement;
+  title: string;
+  displayPopup: boolean;
+}
+
+const ProjectBadge = ({ isApproved, isTopProject, allowDonations }: Props) => {
+  const tCommon = useTranslations('Common');
+  const tProjectDetails = useTranslations('ProjectDetails');
+  const { tenantConfig } = useTenant();
+
+  const badgeConfig: TitleAndIconReturnType | undefined = useMemo(() => {
+    const projectBadge = {
+      notDonatable: {
+        icon: <NewInfoIcon width={10} />,
+        title: tCommon('notDonatable'),
+        displayPopup: true,
+      },
+      topProject: {
+        icon: <TopProjectIcon width={11} />,
+        title: tCommon('topProject'),
+        displayPopup: true,
+      },
+      fieldReviewed: {
+        icon: <FieldReviewedIcon width={10} />,
+        title: tProjectDetails('fieldReviewed'),
+        displayPopup: false,
+      },
+      offSiteReviewed: {
+        icon: <OffSiteReviewedIcon width={10} />,
+        title: tProjectDetails('offSiteReviewed'),
+        displayPopup: false,
+      },
+    };
+    const { notDonatable, topProject, fieldReviewed, offSiteReviewed } =
+      projectBadge;
+    if (!allowDonations) return notDonatable;
+    if (isTopProject && isApproved) return topProject;
+    if (!isTopProject && isApproved) return fieldReviewed;
+    if (!isTopProject && !isApproved) return offSiteReviewed;
+  }, [isApproved, isTopProject, allowDonations, tCommon, tProjectDetails]);
+
+  const getMessage = (title: string | undefined) => {
+    if (title === tCommon('notDonatable')) {
+      return (
+        <div className={style.tooltipContent}>
+          {tenantConfig.config.slug === 'salesforce'
+            ? `${tCommon('salesforceDisabledDonateButtonText')}`
+            : `${tCommon('disabledDonateButtonText')}`}
+        </div>
+      );
+    } else if (title === tCommon('topProject')) {
+      return (
+        <div className={style.tooltipContent}>
+          {tCommon.rich('top_project_standards_fulfilled', {
+            standardsLink: (chunks) => (
+              <a
+                target="_blank"
+                href={tCommon('standardsLink')}
+                rel="noreferrer"
+                className={style.standardsLink}
+                onClick={(e) => e.stopPropagation()}
+              >
+                {chunks}
+              </a>
+            ),
+          })}
+        </div>
+      );
+    }
+    return null;
+  };
+  const { icon, title, displayPopup } = badgeConfig || {};
+  return (
+    <CustomTooltip
+      badgeContent={
+        <div className={style.projectBadge}>
+          <div className={style.badgeIcon}>{icon}</div>
+          <div className={style.badgeTitle}>{title}</div>
+        </div>
+      }
+      shouldDisplayPopup={displayPopup}
+    >
+      {getMessage(title)}
+    </CustomTooltip>
+  );
+};
```

Comment:
1. Separate the `badgeContent` into a separate component.
2. Only enclose the `badgeContent` in `CustomTooltip` when there is a need for a tooltip. i.e. only for top projects in this case.

Reply:
Done